client/sites/components/panel-sidebar/index.tsx

```diff
@@ -21,7 +26,37 @@ export function SidebarItem( { href, children }: { href: string; children: React
 }
 
 export function Sidebar( { children }: { children: ReactNode } ) {
-	return <ul className="panel-sidebar">{ children }</ul>;
+	const isDesktop = useViewportMatch( 'mobile', '>=' );
+	const activeElement = Children.toArray( children ).find(
+		( child ) => isValidElement( child ) && window.location.pathname.startsWith( child.props.href )
+	) as ReactElement;
+
+	if ( isDesktop ) {
+		return (
+			<ul className="panel-sidebar">{ children }</ul>
+		);
+	}
+
+	return (
+		<SelectDropdown
+			className="panel-sidebar panel-sidebar-dropdown"
+			selectedText={ activeElement?.props?.children }
+		>
+			{ Children.toArray( children )
```

Comment:
It doesn't work because of https://github.com/Automattic/wp-calypso/blob/60d9de8844ff7e0f69afb6510f9ba27754fd6b28/packages/components/src/select-dropdown/index.jsx#L159 🥲

Reply:
I mean in mobile case we don't need render the `ul` and just `SelectDropdown.Item` directly?

Reply:
what do you mean by that 😅

Reply:
Chatted via Slack, it won't work because the children is still of type `<SidebarItem/>`.

We can keep current "hacky" way, or revert to `makeSidebar()` style and handle conditional (desktop vs mobile) more cleanly cc: @lsl 

I guess we can keep it this way now, and refactor later.